File: daily/2025/2081_Sum_of_k_Mirror_Numbers.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int num[100];
    bool is_palindrome(long long n, int k){
        int l = 0;
        while(n){
            num[l++] = n % k;
            n /= k;
        }
        l--;
        int i = 0;
        while(i < l){
            if (num[i++] != num[l--]) return false;
        }
        return true;
    }
    long long kMirror(int k, int n) {
        int count = 0;
        long long ans = 0;
        int left = 1;
        for (int i = 1; i < 10; i++){
            if (is_palindrome(i, k)){
                ans += i;
                count++;
                if (count == n) return ans;
            }
        }
        // 10 is not palindrome when base 10
        while(count < n){
            int right = left * 10;
            for (int i = left; i < right; i++){
                long long combined = i;
                int x = i;
                while(x){
                    combined = combined * 10 + (x % 10);
                    x /= 10;
                }
                if (is_palindrome(combined, k)){
                    ans += combined;
                    count++;
                    if (count == n) return ans;
                }
            }
            for (int i = left; i < right; i++){
                for (int j = 0; j < 10; j++){
                    long long combined = i * 10 + j;
                    int x = i;
                    while(x){
                        combined = combined * 10 + (x % 10);
                        x /= 10;
                    }
                    if (is_palindrome(combined, k)){
                        ans += combined;
                        count++;
                        if (count == n) return ans;
                    }
                }
            }
            left *= 10;
        }
        return ans;
    }
};

int main(){
    int k = 2;
    int n = 5;
    // int k = 3;
    // int n = 7;
    // int k = 7;
    // int n = 17;
    Solution S;
    cout << S.kMirror(k, n) << endl;
    return 0;
}

File: daily_python/2081_Sum_of_k_Mirror_Numbers.py
import os
import sys
from typing import List

class Solution:
    def kMirror(self, k: int, n: int) -> int:
        def is_palindrome(x: int) -> bool:
            d = list()
            while x:
                d.append(x % k)
                x //= k
            return d == d[::-1]
        left, count, ans = 1, 0, 0
        while count < n:
            right = left * 10
            for op in [0,1]:
                for i in range(left, right):
                    combined = i
                    x = i if op == 1 else i // 10
                    while x:
                        combined = combined * 10 + (x % 10)
                        x //= 10
                    if is_palindrome(combined):
                        ans += combined
                        count += 1
                        if count == n:
                            return ans
            left = right
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            k = int(lines[0].split('=')[1].strip()[:-1])
            n = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((k, n))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for k, n in dataset:
        results.append(solution.kMirror(k, n))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')